File: modules/my-hello-module/ios/MyHelloModule.m
#import "MyHelloModule.h"

@implementation MyHelloModule

RCT_EXPORT_MODULE(MyHelloModule);

RCT_EXPORT_METHOD(getHello:(RCTPromiseResolveBlock)resolve
                  rejecter:(RCTPromiseRejectBlock)reject)
{
  resolve(@"Hello World from Native (iOS)!");
}

@end

File: modules/my-hello-module/android/src/main/java/com/myhellomodule/MyHelloModule.java
package com.myhellomodule;

import com.facebook.react.bridge.ReactApplicationContext;
import com.facebook.react.bridge.ReactContextBaseJavaModule;
import com.facebook.react.bridge.ReactMethod;

import com.facebook.react.bridge.Promise;

public class MyHelloModule extends ReactContextBaseJavaModule {
    public MyHelloModule(ReactApplicationContext reactContext) {
        super(reactContext);
    }

    @Override
    public String getName() {
        return "MyHelloModule";
    }

    @ReactMethod
    public void getHello(Promise promise) {
        promise.resolve("Hello World from Native!");
    }
}
